clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -2941,6 +2941,69 @@ check_exit_found(void)
     return true;
```

Comment:
@derekbruening When I run the invariant checker on the checked-in context switch trace file, I still see one error:

```

$ bin64/drrun -t drcachesim -simulator_type invariant_checker -infile ../../src/iX-context-switch-inv/clients/drcachesim/tests/mock_switch_sequences.x64.zip -no_abort_on_invariant_error
Trace invariant failure in T3198712 at ref # 22 (2 instrs since timestamp 13347146067972416): Thread is missing exit
Found 1 invariant errors

```

Do we want to add the thread-exit to the context switch trace, or relax the check?

Reply:
How do you generate mock_switch_sequences.x64.zip? I couldn't find a tool, just the test_kernel_switch_sequences test but it doesn't write it out.

Reply:
I think I used a binary editor on a tiny asm trace.

Reply:
Done.

Reply:
Modified the existing mock_switch_sequences to add a thread exit before the footer, on both context switch sequences in the zip.